tests/http/test_17_ssl_use.py

```diff
@@ -173,26 +173,29 @@ def test_17_06_localhost(self, env: Env, proto):
     @staticmethod
     def gen_test_17_07_list():
         tls13_tests = [
-            [None, True],
-            [['TLS_AES_128_GCM_SHA256'], True],
-            [['TLS_AES_256_GCM_SHA384'], False],
-            [['TLS_CHACHA20_POLY1305_SHA256'], True],
-            [['TLS_AES_256_GCM_SHA384',
-              'TLS_CHACHA20_POLY1305_SHA256'], True],
+            ['def', None, True],
+            ['AES128SHA256', ['TLS_AES_128_GCM_SHA256'], True],
+            ['AES128SHA384', ['TLS_AES_256_GCM_SHA384'], False],
+            ['CHACHA20SHA256', ['TLS_CHACHA20_POLY1305_SHA256'], True],
+            ['AES128SHA384+CHACHA20SHA256', ['TLS_AES_256_GCM_SHA384', 'TLS_CHACHA20_POLY1305_SHA256'], True],
         ]
         tls12_tests = [
-            [None, True],
-            [['ECDHE-ECDSA-AES128-GCM-SHA256', 'ECDHE-RSA-AES128-GCM-SHA256'], True],
-            [['ECDHE-ECDSA-AES256-GCM-SHA384', 'ECDHE-RSA-AES256-GCM-SHA384'], False],
-            [['ECDHE-ECDSA-CHACHA20-POLY1305', 'ECDHE-RSA-CHACHA20-POLY1305'], True],
-            [['ECDHE-ECDSA-AES256-GCM-SHA384', 'ECDHE-RSA-AES256-GCM-SHA384',
+            ['def', None, True],
+            ['AES128ish', ['ECDHE-ECDSA-AES128-GCM-SHA256', 'ECDHE-RSA-AES128-GCM-SHA256'], True],
+            ['AES256ish', ['ECDHE-ECDSA-AES256-GCM-SHA384', 'ECDHE-RSA-AES256-GCM-SHA384'], False],
+            ['CHACHA20ish', ['ECDHE-ECDSA-CHACHA20-POLY1305', 'ECDHE-RSA-CHACHA20-POLY1305'], True],
+            ['AES256ish+CHACHA20ish', ['ECDHE-ECDSA-AES256-GCM-SHA384', 'ECDHE-RSA-AES256-GCM-SHA384',
               'ECDHE-ECDSA-CHACHA20-POLY1305', 'ECDHE-RSA-CHACHA20-POLY1305'], True],
         ]
         ret = []
-        for tls_proto in ['TLSv1.3 +TLSv1.2', 'TLSv1.3', 'TLSv1.2']:
-            for [ciphers13, succeed13] in tls13_tests:
-                for [ciphers12, succeed12] in tls12_tests:
-                    ret.append([tls_proto, ciphers13, ciphers12, succeed13, succeed12])
+        for tls_id, tls_proto in {
+                'TLSv1.2+3': 'TLSv1.3 +TLSv1.2',
+                'TLSv1.2': 'TLSv1.3',
```

Comment:
It doesn't look like these mismatched tls_id/tls_proto names are intended. On the next line as well.

Reply:
Ah, right. Just pushed a fix.